dev/AppLifecycle/AppInstance.cpp

```diff
@@ -33,30 +34,37 @@ namespace winrt::Microsoft::Windows::AppLifecycle::implementation
         // Push past the '----' commandline argument prefix.
         argsStart += 4;
 
-        // We explicitly use find_first_of here, so that the resulting data may contain : as a valid character.
-        auto argsEnd = commandLine.find_first_of(L":", argsStart);
+        auto argsEnd = commandLine.find_first_of(L" ", argsStart);
+
+        // Separate the argument from any behind it on the command-line.
+        std::wstring argument;
         if (argsEnd == std::wstring::npos)
         {
-            return { L"", L"" };
+            argument = commandLine.substr(argsStart);
         }
-
-        if (argsStart > argsEnd)
+        else
         {
-            throw std::overflow_error("commandLine");
+            if (argsStart > argsEnd)
```

Comment:
This test will never be true assuming the standard library is working correctly. Should it be reduced to an assertion?